content/en/docs/concepts/cluster-administration/node-autoscaling.md

```diff
@@ -0,0 +1,276 @@
+to the Kubernetes API, autoscalers also need to interact with cloud provider APIs to provision and
+consolidate Nodes. This means that they need to be explicitly integrated with each supported cloud
+provider. The performance and feature set of a given autoscaler can differ between cloud provider
+integrations.
+
+{{< mermaid >}}
+graph TD
+    na[Node autoscaler]
+    k8s[Kubernetes]
+    cp[Cloud Provider]
+
+    k8s --> |get Pods/Nodes|na
+    na --> |drain Nodes|k8s
+    na --> |create/remove VMs|cp
+    cp --> |get VMs|na
+
+    classDef white_on_blue fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff;
+    classDef blue_on_white fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+    class na blue_on_white;
+    class k8s,cp white_on_blue;
+{{</ mermaid >}}
+
+### Autoscaler implementations
+
+Cluster autoscaler and Karpenter are the two Node autoscalers currently sponsored by
+[SIG Autoscaling](https://github.com/kubernetes/community/tree/master/sig-autoscaling).
+
+From the perspective of a cluster user, both autoscalers should provide a similar Node autoscaling
+experience. Both will provision new Nodes for unschedulable Pods, and both will consolidate the
+Nodes that are no longer optimally utilized.
+
+Different autoscalers may also provide features outside the Node autoscaling scope described on this
+page, and those additional features may differ between them.
+
+Consult the sections below, and the linked documentation for the individual autoscalers to decide
+which autoscaler fits your use-case better.
+
+#### Cluster Autoscaler
+
+Cluster Autoscaler adds or removes Nodes to pre-configured Node groups. Node groups generally map to
+some sort of provider-side VM group. A single instance of Cluster Autoscaler can simultaneously
+manage multiple Node groups. When provisioning, Cluster Autoscaler will add Nodes to the group that
+best fits the requests of pending Pods. When consolidating, Cluster Autoscaler always selects
+specific Nodes to remove as opposed to just resizing the underlying VM group.
+
+Additional context:
+
+* Documentation overview:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md
+* Supported cloud providers:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md#faqdocumentation
+* Extensive FAQ: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md
+* Contact: https://github.com/kubernetes/community/tree/master/sig-autoscaling#contact
+
+#### Karpenter
+
+Karpenter auto-provisions VMs based on [NodePool](https://karpenter.sh/docs/concepts/nodepools/)
+configuration provided by the user. Karpenter handles all aspects of node lifecycle, not just
+autoscaling. It works directly with VMs and doesn't rely on provider-side VM groups.
+
+Additional context:
+
+* Documentation: https://karpenter.sh/
+* Supported cloud providers:
+  https://github.com/kubernetes-sigs/karpenter?tab=readme-ov-file#supported-cloudproviders
```

Comment:
We should be wary of linking to that section, because the Kubernetes project doesn't formally promise or provide support for either of the things that page links to.

Reply:
Karpenter, within Kubernetes, is a library you can use to build a node autoscaler.

Reply:
I can't imagine that we'd remove this section, but we might rename it. I think it's fine to link it personally.

Reply:
> because the Kubernetes project doesn't formally promise or provide support for either of the things that page links to

@sftim I'm curious what you mean by this. Are you implying that the K8s project doesn't promise cloudprovider support like this generally?

Reply:
Instead of "Karpenter is a multi-cloud project supported on the following cloud providers", couldn't we say something like:

```
Karpenter Implementations
Karpenter is a multi-cloud project with implementations by the following cloud providers:
```

Reply:
Agreed with this, can you update the phrasing to @chrisnegus's suggested wording please @towca.

We can come back and update this if we start to see community backed implementations.

Reply:
> Are you implying that the K8s project doesn't promise cloudprovider support like this generally?

We support cloud provider integrations if the integration code is part of Kubernetes (for example, we'd triage a bug report and might work on a fix, and security response committee would triage vulnerability reports).

If it's third party - _closed source, or source available but with code outside of the https://github.com/kubernetes/ and https://github.com/kubernetes-sigs/ repositories_ -  we don't promise **any** support. There's an exception for https://github.com/etcd-io/etcd/ which is maintained by a Kubernetes SIG but has its own processes.

Reply:
Linking to https://github.com/kubernetes-sigs/karpenter?tab=readme-ov-file#karpenter-implementations is fine.

Reply:
* Changed "Supported cloud providers" to "Cloud provider integrations" both here and in CA section above.
* The "Implementation comparison" section already covers the differences in how CA and Karpenter integrate with cloud providers.

I think this captures all the suggestions in this thread - WDYT about the current version?